waku/node/jsonrpc/relay/handlers.nim

```diff
@@ -93,18 +94,43 @@ proc installRelayApiHandlers*(node: WakuNode, server: RpcServer, cache: MessageC
         timestamp: msg.timestamp.get(Timestamp(0)),
         ephemeral: msg.ephemeral.get(false)
       )
-    
+
+    # ensure the node is subscribed to the pubsubTopic. otherwise it risks publishing
+    # to a topic with no connected peers
+    if pubsubTopic notin node.wakuRelay.subscribedTopics():
+      raise newException(ValueError, "Failed to publish: Node not subscribed to pubsubTopic: " & pubsubTopic)
+
+    # if RLN is mounted, append the proof to the message
     when defined(rln):
       if not node.wakuRlnRelay.isNil():
-        let success = node.wakuRlnRelay.appendRLNProof(message, 
+        # append the proof to the message
+        let success = node.wakuRlnRelay.appendRLNProof(message,
                                                       float64(getTime().toUnix()))
         if not success:
-          raise newException(ValueError, "Failed to append RLN proof to message")
-
-    let publishFut = node.publish(topic, message)
-
-    if not await publishFut.withTimeout(futTimeout):
-      raise newException(ValueError, "Failed to publish to topic " & topic)
+          raise newException(ValueError, "Failed to publish: error appending RLN proof to message")
+        # validate the message before sending it
+        let result = node.wakuRlnRelay.validateMessage(message)
+        if result == MessageValidationResult.Invalid:
+          raise newException(ValueError, "Failed to publish: invalid RLN proof")
+        elif result == MessageValidationResult.Spam:
+          raise newException(ValueError, "Failed to publish: limit exceeded, try again later")
+        elif result == MessageValidationResult.Valid:
+          debug "Publishing message WITH RLN proof", pubSubTopic=pubSubTopic
+          let publishFut = node.publish(pubsubTopic, message)
+          if not await publishFut.withTimeout(futTimeout):
+            raise newException(ValueError, "Failed to publish: timed out")
+        else:
+          raise newException(ValueError, "Failed to publish: unknown RLN proof validation result")
+      else:
+        raise newException(ValueError, "Failed to publish: RLN enabled but not mounted")
+
+    # if RLN is not mounted, publish the message as is
+    else:
+      debug "Publishing message WITHOUT RLN proof", pubSubTopic=pubSubTopic
+      let publishFut = node.publish(pubsubTopic, message)
+
+      if not await publishFut.withTimeout(futTimeout):
+        raise newException(ValueError, "Failed to publish: timed out")
```

Comment:
I think these `ValueError`'s can be `CatchableError`'s

Reply:
also, maybe we can operate on the same `message` in both cases? i.e rln is enabled, rln is disabled?
then we can save some loc by having just one 
```nim
let publishFut = node.publish(pubsubTopic, message)
if not await publishFut.withTimeout(futTimeout):
    raise newException(ValueError, "Failed to publish: timed out")
```

Reply:
ow indeed, that was a bit spageti.
c964916c2f5260bd79c96ab6a247446fea1a92ba